stash_engine/lib/stash/organization/ror.rb

```diff
@@ -0,0 +1,99 @@
+require 'httparty'
+
+module Stash
+
+  module Organization
+
+    module Ror
+      # Module that facilitates communications with the ROR API:
+      #    https://github.com/ror-community/ror-api/blob/master/api_documentation.md
+      URI = 'https://api.ror.org/organizations'.freeze
+      HEADERS = { 'Content-Type': 'application/json' }.freeze
+      ROR_MAX_RESULTS = 20.0
+      MAX_PAGES = 5
+
+      # Search the ROR API for the given string. This will search name, acronyms, aliases, etc.
+      # @return an Array of Hashes { id: 'https://ror.org/12345', name: 'Sample University' }
+      # The ROR limit appears to be 40 results (even with paging :/)
+      def find_by_ror_name(query)
+        resp = query_ror(URI, { 'query.names': query }, HEADERS)
+        results = process_pages(resp, query) if resp.parsed_response.present? && resp.parsed_response['items'].present?
+        results.present? ? results.flatten.uniq.sort_by { |a| a[:name] } : []
+      end
+
+      # Search ROR and return the first match for the given name
+      # @return a Stash::Organization::Ror::Organization object or nil
+      def find_first_by_ror_name(ror_name)
+        resp = query_ror(URI, { 'query.names': ror_name }, HEADERS)
+        return nil if resp.parsed_response.blank? || resp.parsed_response['items'].blank?
+        result = resp.parsed_response['items'].first
+        return nil if result['id'].blank? || result['name'].blank?
+        find_by_ror_id(result['id'])
+        ror_results_to_hash(resp)
+      end
+
+      # Search the ROR API for a specific organization.
+      # @return a Stash::Organization::Ror::Organization object or nil
+      def find_by_ror_id(ror_id)
+        resp = HTTParty.get("#{URI}/#{ror_id}", headers: HEADERS)
+        return nil if resp.parsed_response.blank? ||
+                      resp.parsed_response['id'].blank? ||
+                      resp.parsed_response['name'].blank?
+        Organization.new(resp.parsed_response)
```

Comment:
After this line you may want to add a "rescue" in case ROR goes down or has problems.

I don't know if I'm rescuing all the possible errors, but I had to add that to Ryan's method identifier#submitter_country since when their API went down it caused failures.

This is probably not as big a deal with AJAX since probably the autocomplete will just stop working instead of causing the problems that Ryan's method caused in the review page.